sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -922,6 +922,22 @@ object SQLConf {
     .checkValues(Set("none", "uncompressed", "snappy", "gzip", "lzo", "lz4", "brotli", "zstd"))
     .createWithDefault("snappy")
 
+  val PARQUET_COMPRESSION_ZSTD_LEVEL = buildConf("spark.sql.parquet.zstd.level")
+    .doc("Sets the zstd level when writing Parquet files and compression codec is `zstd`. " +
+      "The valid range is 1~22. Generally the higher compression level, the higher compression " +
+      "ratio can be achieved, but the writing time will be longer.")
+    .version("3.4.0")
+    .intConf
+    .createWithDefault(3)
```

Comment:
curious how we come to level 3 by default? Any data or benchmark backed for the choice?

Reply:
This is also from Parquet